packages/checkout/widgets-lib/src/widgets/wallet/components/WalletAddress/WalletAddress.tsx

```diff
@@ -1,13 +1,56 @@
-import { Box, Body, Heading } from '@biom3/react';
+import { MenuItem, ButtCon } from '@biom3/react';
 import { Web3Provider } from '@ethersproject/providers';
 import { useEffect, useState } from 'react';
+import { WalletProviderName } from '@imtbl/checkout-sdk';
+import { getWalletLogoByName } from 'lib/logoUtils';
+import { useTranslation } from 'react-i18next';
+import { abbreviateWalletAddress } from 'lib/utils';
 import {
-  walletAddressContainerStyle, walletAddressLayoutStyle, walletAddressTextLayoutStyle, truncatedTextStyle,
-} from './WalletAddressStyles';
-import { CopyButton } from './CopyButton';
+  UserJourney,
+  useAnalytics,
+} from '../../../../context/analytics-provider/SegmentAnalyticsProvider';
 
-export function WalletAddress({ provider }: { provider?: Web3Provider }) {
+const isCopiedStyle = {
+  background: 'base.color.status.success.bright',
+  fill: 'base.color.status.success.bright',
+};
+
+export function WalletAddress({
+  provider,
+  showL1Warning,
+  setShowL1Warning,
+}: {
+  provider?: Web3Provider;
```

Comment:
Can we grab the `provider` from `connectLoaderState` inside this component? Saves having to pass it down from Settings > TransferAssetsL1Warning > WalletAddress

Reply:
tried to use `provider` within `WalletAddress` but had issues using it when `WalletAddress` is used within `TransferAssetsL1Warning`

Reply:
will keep it this way to unblock rewards team